torch/_higher_order_ops/while_loop.py

```diff
@@ -30,6 +193,8 @@ def __call__(
         self,
         cond_fn: Callable,
         body_fn: Callable,
+        body_grad_fn: Callable,
```

Comment:
Ideally, the backward of while_loop should be another while_loop operator with the same function signature `cond_fn`, `body_fn`, `operands`.

To get a backward formula for backward cond and backward body. We  consider the following example.
- forward: Inp0 -> body_fn -> inp1 -> body_fn -> … inpN-1 -> body_fn -> output.
- corresponding backward could be: grad_output, Output, inpN-1-> body_fn_reverse -> grad_inpN-1,  inpN-1, , inpN-2 -> body_fn_reverse -> …inp1, grad_inp1, inp0 -> body_fn_reverse -> grad_inp0

Specifically, one possible design could be:
```python
backward_cond_fn(grad, fw_outputs(0…N)):
  return len(inputs) > 1
```

```python
def backward_body_fn(grad: Tensors, fw_outputs: TensorList):
  Output = fw_outputs.pop() # (0…N-1)
  InpN-1 = fw_outputs.back()
  grad_N-1 = fw_bw(grad, output, inpN-1,) # do a re-computation based on inpN-1 since we only save the outputs of each iteration.
  Return gradN-1, fw_outputs #(0…N-1)
```

The backward is then:
```
while_loop(backward_cond_fn, backward_body_fn, (grad_out, fw_outputs))
```

Reply:
I tried to follow your suggestion in my latest commit.